cmd/doltgres/main.go

```diff
@@ -219,6 +221,10 @@ func main() {
 
 	// TODO: override other aspects of cfg with command line params
 
+	var stop context.CancelFunc
+	ctx, stop = signal.NotifyContext(ctx, os.Interrupt, syscall.SIGTERM)
+	defer stop()
```

Comment:
What is this doing?

Reply:
This is allowing that we catch the `SIGTERM` that actually calls the `server.Stop()` which cleans up everything including the removing the socket file when closing the server. We were not stopping the server properly before it seems (the `Close()` function was not getting called before)